[Teddayz] iP

Teddayz opened the pull request:
### Meeks

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Meeks allows you to focus on the things that are in front of you. It's,

- text-based
- easy to pick up!
- extremely fast to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you and attend to the important matters at hand 😉

Most importantly, it is completely _**FREE**_

Features:

- [ ] Manages tasks
- [ ] Manages deadlines
- [ ] Manages events

Practice Java on Meeks! Here's the `main` method:
```java
import ui.Ui;
import java.util.Scanner;

public class Meeks {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Echo echo = new Echo();
        Ui ui = new Ui();
        ui.greet();
        while (true) {
            String command = scanner.nextLine();
            echo.setWord(command);


            if (command.isEmpty()) {
                System.out.println("Input cannot be empty. Please try again.");
                continue;
            }
            echo.echoOut();

            if (command.equalsIgnoreCase("bye")) {
                break;
            }
        }
        scanner.close();
    }
}
```

j-lum committed:
Merge pull request #3 from j-lum/inline-javafx-tutorials

Inline JavaFX tutorials

j-lum committed:
Update Gradle tutorial to reflect new checkstyle config

j-lum committed:
Fix typo referring to `HelloWorld`

j-lum committed:
Add images to JavaFX tutorial 3

j-lum committed:
Merge pull request #7 from j-lum/tutorial-fix

Address issues #5 and #6

j-lum committed:
Remove references to DukeStub

j-lum committed:
Merge pull request #8 from j-lum/tutorial-fix

Remove references to DukeStub

j-lum committed:
Change indentation in FXML samples to match AB-3

j-lum committed:
Merge pull request #14 from j-lum/fxml-indent

Change indentation in FXML samples to match AB-3

j-lum committed:
JavaFX tutorial: Support cross-platform JARs

The OpenJFX plugin expects applications to be modular and bundled
with jlink, resulting in fat jars that are not cross-platform. Let's
manually include the required dependencies so that shadow can package
them properly.

j-lum committed:
Merge pull request #35 from j-lum/xplatformjfx

JavaFX tutorial: Support cross-platform JARs

jiachen247 committed:
Remove unnecessary override annotation

Remove extra '@Override' at the start of the code example.

j-lum committed:
Merge pull request #41 from jiachen247/patch-2

Remove unnecessary override annotation

Commit:
Remove generic tutorials

Let's remove generic tutorials from this repo as they are not specific
to this project tempalte

Commit:
Add text-ui-test files

Commit:
Add Gradle support

Commit:
Add Gradle support

damithc committed:
README.md: Fix typo of => or

Commit:
Improve runtest.* scripts

* Improve the javac command to work for multiple java files.
* In runtest.bat, delete ACTUAL.TXT only if it exists.

Commit:
README: Update setup instructions to match the latest IDE version

Commit:
README.md: Tweak template text

* Let's replace 'Feature 1' with 'Feature-ABC'
  reason: some students kept the 'Feature 1' in the final UG.
* Let's tweak the template for features and usage
  reasons: to fix grammar, to distinguish between output and outcome

damithc committed:
README: Mention the project language level setting

seanleong339 committed:
.gitignore: Fix ACTUAL.txt -> ACTUAL.TXT  (#73)

text-ui-test/runtest.bat|sh scripts generate a file ACTUAL.TXT.
However, .gitignore uses ACTUAL.txt, which means the generated
file will not be ignored by Git on non-Windows OS.

Let's update .gitignore as ACTUAL.txt -> ACTUAL.TXT

Eclipse-Dominator committed:
Bump gradle and lib version

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
README: Update the JDK version 11 -> 17

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Teddayz committed:
Rename, Greet, Exit

Teddayz committed:
Rename, Greet, Exit

Teddayz committed:
Echo

Teddayz committed:
Echo

Teddayz committed:
Add, List

Teddayz committed:
Add, List

Teddayz committed:
Mark As Done

Teddayz committed:
Mark As Done

Teddayz committed:
ToDos, Events, Deadlines

Teddayz committed:
ToDos, Events, Deadlines

Teddayz committed:
Included semi-automated testing

Teddayz committed:
Included semi-automated testing

Teddayz committed:
Handle Errors

Teddayz committed:
Handle Errors

Teddayz committed:
Delete

Teddayz committed:
Delete

Teddayz committed:
Improved Line Spacing within the chatbot

Teddayz committed:
Improved Line Spacing within the chatbot

Teddayz committed:
Save

Teddayz committed:
Save

Teddayz committed:
Merge branch 'branch-Level-7' into master
Implement Level 7 features and changes.

Teddayz committed:
Merge branch 'branch-Level-7' into master
Implement Level 7 features and changes.

Teddayz committed:
Update chatbot to understand Date and Time for Deadline Tasks

Teddayz committed:
Update chatbot to understand Date and Time for Deadline Tasks

Teddayz committed:
Dates and Times

Teddayz committed:
Dates and Times

Teddayz committed:
Merge branch-Level-8 into master

Teddayz committed:
Merge branch-Level-8 into master

Teddayz committed:
Add Ui and TaskList classes

Teddayz committed:
Add Ui and TaskList classes

Teddayz committed:
Add Parser and Storage

Teddayz committed:
Add Parser and Storage

Teddayz committed:
Separate files to different packages

Teddayz committed:
Separate files to different packages

Teddayz committed:
Merge branch 'master' into add-gradle-support

Teddayz committed:
Merge branch 'master' into add-gradle-support

Teddayz committed:
Add Gradle

Teddayz committed:
Add Gradle

Teddayz committed:
Merge branch 'add-gradle-support'

Teddayz committed:
Merge branch 'add-gradle-support'

Teddayz committed:
Add JUnit Tests

Teddayz committed:
Add JUnit Tests

Teddayz committed:
Add Jar file for Main Class

Teddayz committed:
Add Jar file for Main Class

Teddayz committed:
Add JavaDoc Header Messages

Teddayz committed:
Add JavaDoc Header Messages

Teddayz committed:
Add Indentation for TaskList messages

Teddayz committed:
Add Indentation for TaskList messages

Teddayz committed:
Add find a task by searching for keyword in the task description

Teddayz committed:
Add find a task by searching for keyword in the task description

Teddayz committed:
Merge CodingStandard to master

Teddayz committed:
Merge CodingStandard to master

Teddayz committed:
Change the database file location

Teddayz committed:
Change the database file location

Teddayz committed:
Revert "Add JavaDoc Header Messages"

This reverts commit 3fd3e865ad7a5a239845c29656e8dccfddc3b117.

Teddayz committed:
Revert "Add JavaDoc Header Messages"

This reverts commit 0aa5b8374229a461c797bc465738eb4b887f2283.

Teddayz committed:
Revert "Add Indentation for TaskList messages"

This reverts commit 2058041ccdb8f98263a3b0058f7c8b33a769d458.

Teddayz committed:
Revert "Add Indentation for TaskList messages"

This reverts commit 37dcb66603721c9932e26580d1899642ea3666fe.

Teddayz committed:
Revert to previous version to merge with branch-Level-9

Teddayz committed:
Revert to previous version to merge with branch-Level-9

Teddayz committed:
Merge tag 'Level-9'

Teddayz committed:
Merge tag 'Level-9'

Teddayz committed:
Merge branch 'master' of https://github.com/Teddayz/ip

Teddayz committed:
Merge branch 'master' of https://github.com/Teddayz/ip

Teddayz committed:
Add JavaDoc that was removed during revert

Teddayz committed:
Add JavaDoc that was removed during revert

Teddayz committed:
Add iteration 1 of JavaFx

Teddayz committed:
Add iteration 1 of JavaFx

Teddayz committed:
Add dialog boxes for Mudkip's response

Teddayz committed:
Add dialog boxes for Mudkip's response

Teddayz committed:
Add custom behavior to DialogBox

Teddayz committed:
Add custom behavior to DialogBox

Teddayz committed:
Refactor Mudkip to use FXML

Teddayz committed:
Refactor Mudkip to use FXML

Teddayz committed:
Add logic to my GUI chatbot

Teddayz committed:
Add logic to my GUI chatbot

Teddayz committed:
Merge branch 'branch-Level-10'

Teddayz committed:
Merge branch 'branch-Level-10'

yong-zhuo commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @return a list of tasks objects.
+     * @throws IOException when the file cannot be found.
+     */
+    public TaskList load() throws IOException {
```
Maybe consider naming it as 'loadTasks' for more clarity?

Teddayz replied:
Yep thank you!

yong-zhuo commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        switch(type) {
+            case "T":
+                return new ToDoTask(description, isDone);
+            case "D":
+                if (parts.length < 4) {
+                    System.out.print("File corrupted");
+                    return null;
+                }
+                String deadlineStr = parts[3];
+                // Parse the string in the current inputFormat
+                LocalDateTime deadline = LocalDateTime.parse(deadlineStr, inputFormat);
+                // Convert the string into the desired format for me to create a new Task.
+                return new DeadlineTask(description, isDone, deadline.format(outputFormat));
+            case "E":
+                if (parts.length < 5) {
+                    System.out.print("File corrupted");
+                    return null;
+                }
+                String startStr = parts[3];
+                String endStr = parts[4];
+                // Parse the string in the current inputFormat
+                LocalDateTime start = LocalDateTime.parse(startStr, inputFormat);
+                LocalDateTime end = LocalDateTime.parse(endStr, inputFormat);
+                return new EventTask(description, isDone, start.format(outputFormat).trim(), end.format(outputFormat).trim());
+            default:
+                return null;
+        }
+    }
```
Case clauses should not have indentation and should be aligned with the `switch` keyword. 

There should also be a whitespace between `switch` and `(type)`

Teddayz replied:
Oh right! Thanks for the reminder

yong-zhuo commented on src/main/java/Echo.java:
```diff
@@ -0,0 +1,131 @@
+            String[] parts = description.split(" ", 2);
+            String command = parts[0];
+
+        switch (command) {
```
Nice, this is well structured and adheres to OOP principles. Good job on catching the different possible exceptions that could occur!

Teddayz replied:
Thank you!

yong-zhuo commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+        TaskList tasks = new TaskList();
+        BufferedReader reader = new BufferedReader(new FileReader(filePath));
+        String line;
+        while((line = reader.readLine()) != null) {
```
Should have a white space between `while` and `((line = reader.readLine()) != null)`

Teddayz replied:
Done!

yong-zhuo reviewed (commented):
Overall LGTM 👍!  Code follows OOP principles and is well structured and generally follows the coding standards. Variables adheres to naming convention and test cases are appropriate and well designed. Well done!

darkmoongreatsword commented on src/main/java/Echo.java:
```diff
@@ -0,0 +1,131 @@
+            return ui.invalidCommand();
+        }
+    }
+}
```
Well written. Clear program flow structure that adheres to OOP principles. Exceptions are caught early on, and dealt with well.

darkmoongreatsword commented on src/main/java/Echo.java:
```diff
@@ -0,0 +1,131 @@
+    public String echoOut() {
+
+            String description = word;
+            String[] parts = description.split(" ", 2);
```
Not sure that line 41 is needed. If "description" is only used in this instance, is it safe to use "word" directly?

darkmoongreatsword commented on src/main/java/Echo.java:
```diff
@@ -0,0 +1,131 @@
+            }
+
+        case "delete":
+            int index = Integer.parseInt(parts[1]) - 1;
```
parts[1] may not be an integer. Have you handled this exception when parsing?

darkmoongreatsword commented on src/main/java/Echo.java:
```diff
@@ -0,0 +1,131 @@
+                    return "Please enter a valid task number\n" + ui.line();
+                }
+            }
+        case "unmark":
```
"mark" and "unmark" have similar logic. It might be better to combine the cases, and handle them within the switch-case.

darkmoongreatsword reviewed (commented):
Looks good.
Overall, very readable code, with clear documentation.
There are some minor violations in code quality and standards.
However, it does not affect the readability significantly.

Teddayz committed:
Add Java assertions to handle errors

In order to handle errors meaningfully, assertions were implemented
handle invalid indexing of the Tasklist.

Teddayz committed:
Add Java assertions to handle errors

In order to handle errors meaningfully, assertions were implemented
handle invalid indexing of the Tasklist.

Teddayz committed:
Update code to improve quality

To improve the readability of the code, duplication of code was
abstracted out. Obvious comments were also removed as the code was made
more self-explanatory.

Teddayz committed:
Update code to improve quality

To improve the readability of the code, duplication of code was
abstracted out. Obvious comments were also removed as the code was made
more self-explanatory.

Teddayz committed:
Merge pull request #1 from Teddayz/branch-A-Assertions

Add Java assertions to handle errors

Teddayz committed:
Merge pull request #1 from Teddayz/branch-A-Assertions

Add Java assertions to handle errors

Teddayz committed:
Merge branch 'master' into branch-A-CodeQuality

Teddayz committed:
Merge branch 'master' into branch-A-CodeQuality

Teddayz committed:
Merge pull request #2 from Teddayz/branch-A-CodeQuality

Update code to improve quality

Teddayz committed:
Merge pull request #2 from Teddayz/branch-A-CodeQuality

Update code to improve quality

Teddayz committed:
Add workflow for Continuous Integration

Teddayz committed:
Add workflow for Continuous Integration

Teddayz committed:
Merge branch 'master' of https://github.com/Teddayz/ip

Teddayz committed:
Merge branch 'master' of https://github.com/Teddayz/ip

Teddayz committed:
Update assert logic

Teddayz committed:
Update assert logic

Teddayz committed:
Add C-Update extension feature to chatbot

Teddayz committed:
Add C-Update extension feature to chatbot

Teddayz committed:
Merge branch-BCD-Extension with master branch

Teddayz committed:
Merge branch-BCD-Extension with master branch

Teddayz committed:
Fix long method in Echo class

Teddayz committed:
Fix long method in Echo class

Teddayz committed:
Add more error handling cases

Teddayz committed:
Add more error handling cases

Teddayz committed:
Add User Guide to ReadMe

Teddayz committed:
Add User Guide to ReadMe

Teddayz committed:
Merge master with branch-A-UserGuide

Teddayz committed:
Change filepath to become relative

Teddayz committed:
Update Readme with Command Summary

Teddayz committed:
Update ReadMe on Main page

Teddayz committed:
Merge remote branch 'branch-Level-7' into branch-Level-7

Teddayz committed:
Merge branch 'branch-Level-8' of into branch-Level-8

Teddayz committed:
Merge branch 'branch-A-JavaDoc' of into branch-A-JavaDoc

Teddayz committed:
Merge branch 'branch-A-CodingStandard' of into branch-A-CodingStandard

Teddayz committed:
Merge branch 'branch-Level-9' of into branch-Level-9

Teddayz committed:
Remove Meeks.jar

Teddayz committed:
Remove Meeks.jar

Teddayz committed:
Remove Meeks.jar

Teddayz committed:
Remove Meeks.jar file

Teddayz committed:
Merge branch-Level-7 to master

Teddayz committed:
Resolve merge conflicts of branch-Level-10 with master

Teddayz committed:
Fix bug for gradle file

Teddayz committed:
Merge commit branch-Level-8 with master

Teddayz committed:
Remove Meeks.jar and RandomFolder

Teddayz committed:
Merge commit branch-Level-9 with master

Teddayz committed:
Remove Meeks.jar and RandomFolder

Teddayz committed:
Merge branch-A-Assertions with master

Teddayz committed:
Remove Meeks.jar

Teddayz committed:
Merge branch-A-CodingStandard with master

Teddayz committed:
Remove Meeks.jar from branch-A-JavaDoc

Teddayz committed:
Merge branch-A-JavaDoc with master

Teddayz committed:
Remove Meeks.jar from branch-Level-9

Teddayz committed:
Merge branch-Level-9 with master

Teddayz committed:
Remove Meeks.jar from branch-A-CodeQuality

Teddayz committed:
Merge branch 'branch-A-CodeQuality' of into branch-A-CodeQuality

Teddayz committed:
Remove Meeks.jar from branch-A-CodeQuality

Teddayz committed:
Merge branch-A-CodeQuality with master

Teddayz committed:
Remove Meeks.jar from branch-A-UserGuide

Teddayz committed:
Merge branch-A-UserGuide with master

Teddayz committed:
Edit JavaDoc comment based on code quality feedback
